lib/faraday/response/logger.rb

```diff
@@ -10,11 +10,13 @@ class Response
     # lifecycle to a given Logger object. By default, this logs to STDOUT. See
     # Faraday::Logging::Formatter to see specifically what is logged.
     class Logger < Middleware
+      DEFAULT_OPTIONS = { formatter: Logging::Formatter }.merge(Logging::Formatter::DEFAULT_OPTIONS).freeze
```

Comment:
`Logging::Formatter::DEFAULT_OPTIONS` are already used in the formatter class ([here](https://github.com/lostisland/faraday/blob/main/lib/faraday/logging/formatter.rb#L16)), so we shouldn't need to merge them here as well.

Reply:
@iMacTia The problem is that the default options are validated and only the keys that are present in `DEFAULT_OPTIONS` can be used: https://github.com/lostisland/faraday/blob/main/lib/faraday/middleware.rb#L45

Reply:
Ah good point! That is very much messed up 😄
I wish we had gone with some namespaced option like `formatter_options`, but changing that now would mean making this a breaking change.